sycl/unittests/scheduler/RequiredWGSize.cpp

```diff
@@ -25,7 +25,8 @@ std::array<size_t, 3> IncomingLocalSize = {0, 0, 0};
 std::array<size_t, 3> RequiredLocalSize = {0, 0, 0};
 
 static pi_result redefinedProgramCreate(pi_context, const void *, size_t,
-                                        pi_program *) {
+                                        pi_program *p) {
+  *p = reinterpret_cast<pi_program>(new int[1]);
```

Comment:
why the allocation and de-allocation for these mocks (redefinedProgramCreate, redefinedProgramLink, and redefinedProgramRelase ) ? Do we have code that's checking the program pointer against nullptr?

Reply:
I haven't debugged all the issues related to those tests yet - just wanted to include what I've already found at this moment leaving the rest for a future PR. I think I've seen something that gets improved by doing this.

Also, original PR introducing those mocks had an unaddressed review comment asking for exactly this, so I decided to include that piece in this PR as well.

Reply:
Ok, this is actually unnecessary because of

https://github.com/intel/llvm/blob/5512425eade5871755b01ad8014c1febe67e8e76/sycl/test/Unit/lit.cfg.py#L42-L46